<!doctype html><html lang=ja dir=auto><head><meta charset=utf-8><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=viewport content="width=device-width,initial-scale=1,shrink-to-fit=no"><meta name=robots content="index, follow"><title>pip 実行時に sys.stderr.write(f"ERROR: {exc} ") とエラーが出てpipを実行できないときの対処方法 | 🦅 hurutoriya</title><meta name=keywords content="python"><meta name=description content="新しいマシンでpipをセットアップして実行しようとすると
1 2 3 4 5 6 7 8 9 10 11 12 13 14  >> pip Traceback (most recent call last): File &#34;/usr/local/bin/pip&#34;, line 11, in <module> load_entry_point('pip==21.0', 'console_scripts', 'pip')() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 489, in load_entry_point return get_distribution(dist).load_entry_point(group, name) File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2843, in load_entry_point return ep.load() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2434, in load return self.resolve() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2440, in resolve module = __import__(self.module_name, fromlist=['__name__'], level=0) File &#34;/Library/Python/2.7/site-packages/pip-21.0-py2.7.egg/pip/_internal/cli/main.py&#34;, line 60 sys."><meta name=author content="Shunya Ueta"><link rel=canonical href=https://shunyaueta.com/posts/2021-02-08/><link crossorigin=anonymous href=/assets/css/stylesheet.abc7c82c3d415a6df50430738d1cbcc4c76fea558bc5a0c830d3babf78167a35.css integrity="sha256-q8fILD1BWm31BDBzjRy8xMdv6lWLxaDIMNO6v3gWejU=" rel="preload stylesheet" as=style><script defer crossorigin=anonymous src=/assets/js/highlight.f413e19d0714851f6474e7ee9632408e58ac146fbdbe62747134bea2fa3415e0.js integrity="sha256-9BPhnQcUhR9kdOfuljJAjlisFG+9vmJ0cTS+ovo0FeA=" onload=hljs.initHighlightingOnLoad();></script><link rel=icon href=https://shunyaueta.com/favicon.ico><link rel=icon type=image/png sizes=16x16 href=https://shunyaueta.com/favicon-16x16.png><link rel=icon type=image/png sizes=32x32 href=https://shunyaueta.com/favicon-32x32.png><link rel=apple-touch-icon href=https://shunyaueta.com/apple-touch-icon.png><link rel=mask-icon href=https://shunyaueta.com/safari-pinned-tab.svg><meta name=theme-color content="#2e2e33"><meta name=msapplication-TileColor content="#2e2e33"><noscript><style>#theme-toggle,.top-link{display:none}</style><style>@media(prefers-color-scheme:dark){:root{--theme: rgb(29, 30, 32);--entry: rgb(46, 46, 51);--primary: rgb(218, 218, 219);--secondary: rgb(155, 156, 157);--tertiary: rgb(65, 66, 68);--content: rgb(196, 196, 197);--hljs-bg: rgb(46, 46, 51);--code-bg: rgb(55, 56, 62);--border: rgb(51, 51, 51)}.list{background:var(--theme)}.list:not(.dark)::-webkit-scrollbar-track{background:0 0}.list:not(.dark)::-webkit-scrollbar-thumb{border-color:var(--theme)}}</style></noscript><script async src="https://www.googletagmanager.com/gtag/js?id=G-JMJRQJT0Q3"></script><script>window.dataLayer=window.dataLayer||[];function gtag(){dataLayer.push(arguments);}
gtag('js',new Date());gtag('config','G-JMJRQJT0Q3');</script><meta property="og:title" content="pip 実行時に sys.stderr.write(f&#34;ERROR: {exc} &#34;) とエラーが出てpipを実行できないときの対処方法"><meta property="og:description" content="新しいマシンでpipをセットアップして実行しようとすると
1 2 3 4 5 6 7 8 9 10 11 12 13 14  >> pip Traceback (most recent call last): File &#34;/usr/local/bin/pip&#34;, line 11, in <module> load_entry_point('pip==21.0', 'console_scripts', 'pip')() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 489, in load_entry_point return get_distribution(dist).load_entry_point(group, name) File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2843, in load_entry_point return ep.load() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2434, in load return self.resolve() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2440, in resolve module = __import__(self.module_name, fromlist=['__name__'], level=0) File &#34;/Library/Python/2.7/site-packages/pip-21.0-py2.7.egg/pip/_internal/cli/main.py&#34;, line 60 sys."><meta property="og:type" content="article"><meta property="og:url" content="https://shunyaueta.com/posts/2021-02-08/"><meta property="og:image" content="https://shunyaueta.com/ogp.jpg"><meta property="article:section" content="posts"><meta property="article:published_time" content="2021-02-08T23:29:12+09:00"><meta property="article:modified_time" content="2022-10-24T10:10:40+09:00"><meta property="og:site_name" content="Shunya Ueta"><meta name=twitter:card content="summary_large_image"><meta name=twitter:image content="https://shunyaueta.com/ogp.jpg"><meta name=twitter:title content="pip 実行時に sys.stderr.write(f&#34;ERROR: {exc} &#34;) とエラーが出てpipを実行できないときの対処方法"><meta name=twitter:description content="新しいマシンでpipをセットアップして実行しようとすると
1 2 3 4 5 6 7 8 9 10 11 12 13 14  >> pip Traceback (most recent call last): File &#34;/usr/local/bin/pip&#34;, line 11, in <module> load_entry_point('pip==21.0', 'console_scripts', 'pip')() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 489, in load_entry_point return get_distribution(dist).load_entry_point(group, name) File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2843, in load_entry_point return ep.load() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2434, in load return self.resolve() File &#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;, line 2440, in resolve module = __import__(self.module_name, fromlist=['__name__'], level=0) File &#34;/Library/Python/2.7/site-packages/pip-21.0-py2.7.egg/pip/_internal/cli/main.py&#34;, line 60 sys."><script type=application/ld+json>{"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":2,"name":"pip 実行時に sys.stderr.write(f\"ERROR: {exc} \") とエラーが出てpipを実行できないときの対処方法","item":"https://shunyaueta.com/posts/2021-02-08/"}]}</script><script type=application/ld+json>{"@context":"https://schema.org","@type":"BlogPosting","headline":"pip 実行時に sys.stderr.write(f\"ERROR: {exc} \") とエラーが出てpipを実行できないときの対処方法","name":"pip 実行時に sys.stderr.write(f\x22ERROR: {exc} \x22) とエラーが出てpipを実行できないときの対処方法","description":"新しいマシンでpipをセットアップして実行しようとすると\n1 2 3 4 5 6 7 8 9 10 11 12 13 14  \u0026gt;\u0026gt; pip Traceback (most recent call last): File \u0026#34;/usr/local/bin/pip\u0026#34;, line 11, in \u0026lt;module\u0026gt; load_entry_point(\u0026#39;pip==21.0\u0026#39;, \u0026#39;console_scripts\u0026#39;, \u0026#39;pip\u0026#39;)() File \u0026#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py\u0026#34;, line 489, in load_entry_point return get_distribution(dist).load_entry_point(group, name) File \u0026#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py\u0026#34;, line 2843, in load_entry_point return ep.load() File \u0026#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py\u0026#34;, line 2434, in load return self.resolve() File \u0026#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py\u0026#34;, line 2440, in resolve module = __import__(self.module_name, fromlist=[\u0026#39;__name__\u0026#39;], level=0) File \u0026#34;/Library/Python/2.7/site-packages/pip-21.0-py2.7.egg/pip/_internal/cli/main.py\u0026#34;, line 60 sys.","keywords":["python"],"articleBody":"新しいマシンでpipをセットアップして実行しようとすると\n1 2 3 4 5 6 7 8 9 10 11 12 13 14   pip Traceback (most recent call last): File \"/usr/local/bin/pip\", line 11, in  load_entry_point('pip==21.0', 'console_scripts', 'pip')() File \"/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py\", line 489, in load_entry_point return get_distribution(dist).load_entry_point(group, name) File \"/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py\", line 2843, in load_entry_point return ep.load() File \"/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py\", line 2434, in load return self.resolve() File \"/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py\", line 2440, in resolve module = __import__(self.module_name, fromlist=['__name__'], level=0) File \"/Library/Python/2.7/site-packages/pip-21.0-py2.7.egg/pip/_internal/cli/main.py\", line 60 sys.stderr.write(f\"ERROR: {exc}\")   というエラーが出た。\nデフォルトの Python の実行ランタイムが 2.x 系なのでそれに起因するエラーだった。 pip を Python2 系でおそらくインストールしており、pip21 に更新後は Python3.6 からの機能である f-string が利用され始めたことが起因。\n対処方法 既存の pip を削除した上で、\n1 2  curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py python3 get-pip.py   で pip が実行できるようになりエラーが解消された\nReferences  pip install failing on python2  ","wordCount":"109","inLanguage":"ja","datePublished":"2021-02-08T23:29:12+09:00","dateModified":"2022-10-24T10:10:40+09:00","author":{"@type":"Person","name":"Shunya Ueta"},"mainEntityOfPage":{"@type":"WebPage","@id":"https://shunyaueta.com/posts/2021-02-08/"},"publisher":{"@type":"Organization","name":"🦅 hurutoriya","logo":{"@type":"ImageObject","url":"https://shunyaueta.com/favicon.ico"}}}</script></head><body id=top><script>if(localStorage.getItem("pref-theme")==="dark"){document.body.classList.add('dark');}else if(localStorage.getItem("pref-theme")==="light"){document.body.classList.remove('dark')}else if(window.matchMedia('(prefers-color-scheme: dark)').matches){document.body.classList.add('dark');}</script><header class=header><nav class=nav><div class=logo><a href=https://shunyaueta.com/ accesskey=h title="🦅 hurutoriya (Alt + H)">🦅 hurutoriya</a><div class=logo-switches><button id=theme-toggle accesskey=t title="(Alt + T)"><svg id="moon" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24" fill="none" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z"/></svg><svg id="sun" xmlns="http://www.w3.org/2000/svg" width="24" height="18" viewBox="0 0 24 24" fill="none" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="12" cy="12" r="5"/><line x1="12" y1="1" x2="12" y2="3"/><line x1="12" y1="21" x2="12" y2="23"/><line x1="4.22" y1="4.22" x2="5.64" y2="5.64"/><line x1="18.36" y1="18.36" x2="19.78" y2="19.78"/><line x1="1" y1="12" x2="3" y2="12"/><line x1="21" y1="12" x2="23" y2="12"/><line x1="4.22" y1="19.78" x2="5.64" y2="18.36"/><line x1="18.36" y1="5.64" x2="19.78" y2="4.22"/></svg></button></div></div><ul id=menu><li><a href=https://shunyaueta.com/archives title=Archive><span>Archive</span></a></li><li><a href=https://shunyaueta.com/about title=About><span>About</span></a></li><li><a href=https://shunyaueta.com/tags/newsletter/ title=Newsletter><span>Newsletter</span></a></li><li><a href=https://shunyaueta.com/index.xml title=RSS><span>RSS</span></a></li></ul></nav></header><main class=main><article class=post-single><header class=post-header><h1 class=post-title>pip 実行時に sys.stderr.write(f"ERROR: {exc} ") とエラーが出てpipを実行できないときの対処方法</h1><div class=post-meta><span title="2021-02-08 23:29:12 +0900 +0900">February 8, 2021</span>&nbsp;·&nbsp;Shunya Ueta</div></header><div class=post-content><p>新しいマシンで<code>pip</code>をセットアップして実行しようとすると</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt> 1
</span><span class=lnt> 2
</span><span class=lnt> 3
</span><span class=lnt> 4
</span><span class=lnt> 5
</span><span class=lnt> 6
</span><span class=lnt> 7
</span><span class=lnt> 8
</span><span class=lnt> 9
</span><span class=lnt>10
</span><span class=lnt>11
</span><span class=lnt>12
</span><span class=lnt>13
</span><span class=lnt>14
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-bash data-lang=bash>&gt;&gt; pip
Traceback <span class=o>(</span>most recent call last<span class=o>)</span>:
  File <span class=s2>&#34;/usr/local/bin/pip&#34;</span>, line 11, in &lt;module&gt;
    load_entry_point<span class=o>(</span><span class=s1>&#39;pip==21.0&#39;</span>, <span class=s1>&#39;console_scripts&#39;</span>, <span class=s1>&#39;pip&#39;</span><span class=o>)()</span>
  File <span class=s2>&#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;</span>, line 489, in load_entry_point
    <span class=k>return</span> get_distribution<span class=o>(</span>dist<span class=o>)</span>.load_entry_point<span class=o>(</span>group, name<span class=o>)</span>
  File <span class=s2>&#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;</span>, line 2843, in load_entry_point
    <span class=k>return</span> ep.load<span class=o>()</span>
  File <span class=s2>&#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;</span>, line 2434, in load
    <span class=k>return</span> self.resolve<span class=o>()</span>
  File <span class=s2>&#34;/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources/__init__.py&#34;</span>, line 2440, in resolve
    <span class=nv>module</span> <span class=o>=</span> __import__<span class=o>(</span>self.module_name, <span class=nv>fromlist</span><span class=o>=[</span><span class=s1>&#39;__name__&#39;</span><span class=o>]</span>, <span class=nv>level</span><span class=o>=</span>0<span class=o>)</span>
  File <span class=s2>&#34;/Library/Python/2.7/site-packages/pip-21.0-py2.7.egg/pip/_internal/cli/main.py&#34;</span>, line <span class=m>60</span>
    sys.stderr.write<span class=o>(</span>f<span class=s2>&#34;ERROR: {exc}&#34;</span><span class=o>)</span>
</code></pre></td></tr></table></div></div><p>というエラーが出た。</p><p>デフォルトの Python の実行ランタイムが 2.x 系なのでそれに起因するエラーだった。
pip を Python2 系でおそらくインストールしており、pip21 に更新後は Python3.6 からの機能である f-string が利用され始めたことが起因。</p><h2 id=対処方法>対処方法<a hidden class=anchor aria-hidden=true href=#対処方法>#</a></h2><p>既存の pip を削除した上で、</p><div class=highlight><div class=chroma><table class=lntable><tr><td class=lntd><pre class=chroma><code><span class=lnt>1
</span><span class=lnt>2
</span></code></pre></td><td class=lntd><pre class=chroma><code class=language-bash data-lang=bash>curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
python3 get-pip.py
</code></pre></td></tr></table></div></div><p>で pip が実行できるようになりエラーが解消された</p><h2 id=references>References<a hidden class=anchor aria-hidden=true href=#references>#</a></h2><ul><li><a href=https://stackoverflow.com/questions/65866417/pip-install-failing-on-python2>pip install failing on python2</a></li></ul><h2>See Also</h2><ul><li><a href=/posts/2020-12-26/>PythonでApache beam 入門</a></li><li><a href=/posts/2020-09-09/>pandas を使って特定のディレクトリのCSVファイルをすべて連結して一つのCSVファイルを作成</a></li><li><a href=/posts/2020-08-23/>Python の内包表記とジェネレータ式のメモリ使用量比較</a></li><li><a href=/posts/2020-08-04/>AOJの「ITP I」40問をPythonで解いた</a></li><li><a href=/posts/2020-07-25/>How to write the UnitTest with stdin at Pytest</a></li></ul><h2>Support</h2>記事をお読みくださりありがとうございます。このウェブサイトの運営を支援していただける方を募集しています。
もしよろしければ、下のボタンからサポート(投げ銭)していただけると、ブログ執筆、情報発信のモチベーションに繋がります✨
<a href=https://www.buymeacoffee.com/hurutoriya><img src="https://img.buymeacoffee.com/button-api/?text=Buy me a coffee&emoji=☕&slug=hurutoriya&button_colour=FFDD00&font_colour=000000&font_family=Inter&outline_colour=000000&coffee_colour=ffffff"></a></div><footer class=post-footer><ul class=post-tags><li><a href=https://shunyaueta.com/tags/python/>python</a></li></ul><nav class=paginav><a class=prev href=https://shunyaueta.com/posts/2021-02-09/><span class=title>« 前のページ</span><br><span>Standard SQLで 列と列の組み合わせの数を集計したい</span></a>
<a class=next href=https://shunyaueta.com/posts/2021-01-17/><span class=title>次のページ »</span><br><span>TFXの歴史を振り返りつつ機械学習エンジニアリングを提案する論文「Towards ML Engineering: A Brief History Of TensorFlow Extended (TFX)」</span></a></nav></footer><script src=https://giscus.app/client.js data-repo=hurutoriya/hurutoriya.github.io data-repo-id="MDEwOlJlcG9zaXRvcnk2MDgyNTY1Nw==" data-category=Comments data-category-id=DIC_kwDOA6AgOc4CAQTX data-mapping=pathname data-reactions-enabled=1 data-emit-metadata=0 data-theme=dark data-lang=ja crossorigin=anonymous async></script></article></main><footer class=footer><span>&copy; 2022 <a href=https://shunyaueta.com/>🦅 hurutoriya</a></span>
<span>Powered by
<a href=https://gohugo.io/ rel="noopener noreferrer" target=_blank>Hugo</a> &
        <a href=https://github.com/adityatelange/hugo-PaperMod/ rel=noopener target=_blank>PaperMod</a></span></footer><a href=#top aria-label="go to top" title="Go to Top (Alt + G)" class=top-link id=top-link accesskey=g><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 6" fill="currentcolor"><path d="M12 6H0l6-6z" /></svg></a><script>let menu=document.getElementById('menu')
if(menu){menu.scrollLeft=localStorage.getItem("menu-scroll-position");menu.onscroll=function(){localStorage.setItem("menu-scroll-position",menu.scrollLeft);}}
document.querySelectorAll('a[href^="#"]').forEach(anchor=>{anchor.addEventListener("click",function(e){e.preventDefault();var id=this.getAttribute("href").substr(1);if(!window.matchMedia('(prefers-reduced-motion: reduce)').matches){document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView({behavior:"smooth"});}else{document.querySelector(`[id='${decodeURIComponent(id)}']`).scrollIntoView();}
if(id==="top"){history.replaceState(null,null," ");}else{history.pushState(null,null,`#${id}`);}});});</script><script>var mybutton=document.getElementById("top-link");window.onscroll=function(){if(document.body.scrollTop>800||document.documentElement.scrollTop>800){mybutton.style.visibility="visible";mybutton.style.opacity="1";}else{mybutton.style.visibility="hidden";mybutton.style.opacity="0";}};</script><script>document.getElementById("theme-toggle").addEventListener("click",()=>{if(document.body.className.includes("dark")){document.body.classList.remove('dark');localStorage.setItem("pref-theme",'light');}else{document.body.classList.add('dark');localStorage.setItem("pref-theme",'dark');}})</script><script>document.querySelectorAll('pre > code').forEach((codeblock)=>{const container=codeblock.parentNode.parentNode;const copybutton=document.createElement('button');copybutton.classList.add('copy-code');copybutton.innerHTML='コピー';function copyingDone(){copybutton.innerHTML='コピーされました!';setTimeout(()=>{copybutton.innerHTML='コピー';},2000);}
copybutton.addEventListener('click',(cb)=>{if('clipboard'in navigator){navigator.clipboard.writeText(codeblock.textContent);copyingDone();return;}
const range=document.createRange();range.selectNodeContents(codeblock);const selection=window.getSelection();selection.removeAllRanges();selection.addRange(range);try{document.execCommand('copy');copyingDone();}catch(e){};selection.removeRange(range);});if(container.classList.contains("highlight")){container.appendChild(copybutton);}else if(container.parentNode.firstChild==container){}else if(codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.nodeName=="TABLE"){codeblock.parentNode.parentNode.parentNode.parentNode.parentNode.appendChild(copybutton);}else{codeblock.parentNode.appendChild(copybutton);}});</script></body></html>